homeassistant/components/homekit/type_media_players.py

```diff
@@ -142,3 +170,182 @@ def update_state(self, new_state):
                               self.entity_id, current_state)
                 self.chars[FEATURE_TOGGLE_MUTE].set_value(current_state)
             self._flag[FEATURE_TOGGLE_MUTE] = False
+
+
+@TYPES.register('TelevisionMediaPlayer')
+class TelevisionMediaPlayer(HomeAccessory):
+    """Generate a Television Media Player accessory."""
+
+    def __init__(self, *args):
+        """Initialize a Switch accessory object."""
+        super().__init__(*args, category=CATEGORY_TELEVISION)
+
+        self._flag_active = False
+        self._flag_active_identifier = False
+        self._flag_toggle_mute = False
+
+        # Add additional characteristics if volume or input selection supported
+        self.chars = []
```

Comment:
So the reason why I didn't do this is because `ActiveIdentifier` is a required characteristic by the Television accessory. I believe it will cause errors to add this to the array passed to `add_preload_service `. Somehow, we need to be able to track the available features (select a source) for the entity in HA, but not conflict with the required characteristics of the Television accessory. I do not see any negative side effect of adding these services to the accessory, even they go unused. The `RemoteKey` char should definitely be added since it's optional.

Reply:
Never mind :) saw the comment below.